### ISSUE TITLE
Accuracy metric incorrect for hinge loss objective

### ISSUE THREAD

**Issue by around1991** [fchollet/keras#3372]
In the metrics module, the way binary accuracy is calculated assumes your predictions and labels are 0/1. However, if you're using hinge loss, your training labels are -1/1, and so the accuracy metric will give incorrect results.

It's easy to work around (define a custom metric to use), but it is slightly inconvenient to the user who assumes that binary accuracy will automatically do the right thing. Is there any way the binary accuracy metric can be a bit cleverer and look at the loss that's used?

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by RaffEdwardBAH** [comment]
@fchollet Just ran into this issue I think. A fix or at least mention in the documentation would be enormously helpful. As is its hard to tell what is going on.

**Comment by sonney2k** [comment]
Please reopen!

**Comment by Al-Sad** [comment]
Thank you for pointing out this problem

**Comment by BorjaEst** [comment]
Please reopen!
